core/src/main/java/org/apache/iceberg/rest/ResourcePaths.java

```diff
@@ -40,39 +46,48 @@ public static String tokens() {
   }
 
   private final String prefix;
+  private final String namespaceSeparator;
 
   public ResourcePaths(String prefix) {
+    this(prefix, RESTUtil.NAMESPACE_ESCAPED_SEPARATOR);
```

Comment:
Should we deprecate this constructor so that `forCatalogProperties` is the only option in the future?

Reply:
yep, good point. I don't think this constructor was used anywhere so it's good to deprecate and guide users to use `forCatalogProperties´ directly